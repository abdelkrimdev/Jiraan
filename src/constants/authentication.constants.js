export const authenticationConstants = {
    // User Sign Up
    USER_CREATE_REQUEST: 'FETCH_CREATE_USER_REQUEST',
    USER_CREATE_SUCCESS: 'FETCH_CREATE_USER_SUCCESS',
    USER_CREATE_FAILURE: 'FETCH_CREATE_USER_FAILURE',
    // User Sign In
    USER_LOGIN_REQUEST: 'FETCH_LOGIN_USER_REQUEST',
    USER_LOGIN_SUCCESS: 'FETCH_LOGIN_USER_SUCCESS',
    USER_LOGIN_FAILURE: 'FETCH_LOGIN_USER_FAILURE',
    // Get Current User
    USER_CURRENT_ACTION: 'FIRE_CURRENT_USER_ACTION',
    // User Sign Out
    USER_LOGOUT_ACTION: 'FIRE_LOGOUT_USER_ACTION'
};
