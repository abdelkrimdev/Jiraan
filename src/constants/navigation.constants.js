export const navigationConstants = {
  // Switch Navigator Keys
  CHECK: 'Check',
  AUTH: 'Auth',
  APP: 'App',
  // Authentication Stack Keys
  LOGIN: 'Login'
}
