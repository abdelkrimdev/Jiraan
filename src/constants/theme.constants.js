export const themeConstants = {
  // Theme Colors
  PRIMARY_COLOR: '#8e44ad',
  SECONDARY_COLOR: '#95a5a6',
  SUCCESS_COLOR: '#27ae60',
  DANGER_COLOR: '#e74c3c',
  WARNING_COLOR: '#f1c40f',
  INFO_COLOR: '#3498db',
  LIGHT_COLOR: '#ecf0f1',
  DARK_COLOR: '#2c3e50'
}
