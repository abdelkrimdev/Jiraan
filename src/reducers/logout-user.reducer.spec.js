import { authenticationConstants } from '../constants/authentication.constants'
import { logoutUser } from './logout-user.reducer'

describe('logout user reducer', () => {
  it('should have initial state.', () => {
    const action = { type: 'DUMMY_ACTION' }

    expect(logoutUser(undefined, action)).toMatchSnapshot()
  })

  it('should return the current state after receiving a dummy action.', () => {
    const state = { data: 'current state' }
    const action = { type: 'DUMMY_ACTION' }

    expect(logoutUser(state, action)).toMatchSnapshot()
  })

  it('should handle logout user request.', () => {
    const action = { type: authenticationConstants.USER_LOGOUT_REQUEST }

    expect(logoutUser({ }, action)).toMatchSnapshot()
  })

  it('should handle logout user success.', () => {
    const state = { fetching: true }
    const action = {
      type: authenticationConstants.USER_LOGOUT_SUCCESS,
      user: {
        displayName: 'username',
        email: 'user@somewhere.com'
      }
    }

    expect(logoutUser(state, action)).toMatchSnapshot()
  })

  it('should handle logout user failure.', () => {
    const state = { fetching: true }
    const action = {
      type: authenticationConstants.USER_LOGOUT_FAILURE,
      error: {
        code: 'error/code',
        message: 'Logout User Error Message'
      }
    }

    expect(logoutUser(state, action)).toMatchSnapshot()
  })
})
