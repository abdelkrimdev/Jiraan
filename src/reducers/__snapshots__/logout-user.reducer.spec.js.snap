// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logout user reducer should handle logout user failure. 1`] = `
Object {
  "error": Object {
    "code": "error/code",
    "message": "logout User Error Message",
  },
  "fetching": false,
}
`;

exports[`logout user reducer should handle logout user request. 1`] = `
Object {
  "fetching": true,
}
`;

exports[`logout user reducer should handle logout user success. 1`] = `
Object {
  "fetching": false,
}
`;

exports[`logout user reducer should have initial state. 1`] = `Object {}`;

exports[`logout user reducer should return the current state after receiving a dummy action. 1`] = `
Object {
  "data": "current state",
}
`;
